synapse/lib/view.py

```diff
@@ -1254,6 +1254,89 @@ async def _calcChildViews(self):
 
                 todo.append(child)
 
+    async def insertParentFork(self, useriden, name=None):
+        '''
+        Insert a new View between a forked View and its parent.
+
+        Returns:
+            New view definition with the same perms as the current fork.
+        '''
+        if not self.isafork():
+            mesg = f'View ({self.iden}) is not a fork, cannot insert a new fork between parent.'
+            raise s_exc.BadState(mesg=mesg)
+
+        ctime = s_common.now()
+        layriden = s_common.guid()
+
+        ldef = {
+            'iden': layriden,
+            'created': ctime,
+            'creator': useriden,
+            'lockmemory': self.core.conf.get('layers:lockmemory'),
+            'logedits': self.core.conf.get('layers:logedits'),
+            'readonly': False
+        }
+
+        vdef = {
+            'iden': s_common.guid(),
+            'created': ctime,
+            'creator': useriden,
+            'parent': self.parent.iden,
+            'layers': [layriden] + [lyr.iden for lyr in self.parent.layers]
+        }
+
+        if name is not None:
+            vdef['name'] = name
+
+        s_layer.reqValidLdef(ldef)
+        s_schemas.reqValidView(vdef)
+
+        return await self._push('view:forkparent', ldef, vdef)
+
+    @s_nexus.Pusher.onPush('view:forkparent')
+    async def _insertParentFork(self, ldef, vdef):
+
+        s_layer.reqValidLdef(ldef)
+        s_schemas.reqValidView(vdef)
+
+        await self.core._addLayer(ldef, (None, None))
+        await self.core._addView(vdef)
+
+        forkiden = vdef.get('iden')
+        self.parent = self.core.getView(forkiden)
```

Comment:
```suggestion
        self.parent = self.core.reqView(forkiden)
```
Is it possible for the parent view to have disappeared?

Reply:
Shouldn't be possible but it doesn't hurt to require it here.